### PR TITLE
move onto AndroidX

### PR DESCRIPTION
**Pull request by hnljp**
Migrated from the deprecated Android Support libraries to the newest versions of the AndroidX libraries.
The Android gradle plugin is updated to the current 4.0.0.
Glide is updated to the current 4.11.0, and 4.1x.x is needed for AndroidX support.

### CODE REVIEW

**Comment by hnljp** [comment]
@iainmcgin In case you are not able to get the OpenID CLA for https://github.com/openid/AppAuth-Android/pull/508 have I created my own pullrequest.
All dependencies (except the test ones) have been updated and I am supporting Android API 30.
Have also added a comment on getting AppAuth to work on Android 11.

**Comment by hnljp** [comment]
This PR documents Android 11 support: https://github.com/openid/AppAuth-Android/issues/554
Fixes:
- https://github.com/openid/AppAuth-Android/issues/417
- https://github.com/openid/AppAuth-Android/issues/423
- https://github.com/openid/AppAuth-Android/issues/425

File: README.md [hunk 1]
```diff
@@ -51,6 +51,12 @@ either through custom URI scheme redirects, or App Links.
 AS's that assume all clients are web-based or require clients to maintain
 confidentiality of the client secrets may not work well.
 
+From Android API 30 (R) and above, set [QUERY_ALL_PACKAGES](https://developer.android.com/reference/kotlin/android/Manifest.permission#query_all_packages) in the manifest,
```
Comment (M66B) [inline README.md:54]: This permission will most likely not be allowed for common apps (not launcher, not security related, etc) in the Play store. An alternative might be:

```
    <queries>
        <intent>
            <action android:name="android.intent.action.VIEW" />
            <data android:mimeType="*/*" />
        </intent>
    </queries>
```

However, it remains to be seen if Google will allow this for apps published in the Play store.

Comment (hnljp) [reply README.md]: Updated in the newest commit, limited to what we actually need.
See https://github.com/openid/AppAuth-Android/issues/554

File: app/AndroidManifest.xml [hunk 2]
```diff
@@ -16,7 +16,14 @@
         package="net.openid.appauthdemo" >
 
     <uses-permission android:name="android.permission.INTERNET" />
-    <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" />
+
+    <queries>
```
Comment (M66B) [inline app/AndroidManifest.xml:20]: I am not sure if this is a good idea because https://github.com/openid/AppAuth-Android/pull/469

Comment (hnljp) [reply app/AndroidManifest.xml:20]: is there any documentation on Opera, would be nice with a list of supported categories by Opera.
I would actually think that it should instead be a bug report to Opera, as it should support android.intent.category.BROWSABLE.
It has even been around for a long time... https://developer.android.com/reference/android/content/Intent#CATEGORY_BROWSABLE

Comment (M66B) [reply app/AndroidManifest.xml:20]: I don't know. I don't use Opera.

Comment (hnljp) [reply app/AndroidManifest.xml:20]: Seems Opera is supporting https://developer.android.com/reference/android/content/Intent#CATEGORY_APP_BROWSER
Will allow both and add a check for it in the BrowserSelector.

Comment (hnljp) [reply app/AndroidManifest.xml:20]: Not all browsers seem to support it, so I am still letting it also check for https://developer.android.com/reference/android/content/Intent#CATEGORY_BROWSABLE

**Comment by blundell** [comment]
@hnljp it's a bit cold to just take someones changes contributed in an effort to help an oss project and reauthor them as your own. I've signed all the CLA's its just a waiting game.

**Comment by hnljp** [comment]
> @hnljp it's a bit cold to just take someones changes contributed in an effort to help an oss project and reauthor them as your own. I've signed all the CLA's its just a waiting game.
I did not copy your code, I made a new pull request from the ground.
I am happy that you have signed to CLA's, so I will close the this pull request.

**Pull request closed without merging**